include/filewriter/summarywriter_impl.hpp

```diff
@@ -150,6 +150,120 @@ void SummaryWriter<Filewriter>::Image(const std::string& tag,
     remove(fileNames[i].c_str());
 }
 
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::Embedding(
+    const std::string &tensorName,
+    const std::string &tensordataPath,
+    Filewriter& fw,
+    const std::string &metadataPath,
+    const std::vector<size_t> &tensorShape)
+{
+    mlboard::SummaryMetadata_PluginData *pluginData = new SummaryMetadata::PluginData();
+    pluginData->set_plugin_name("projector");
+    mlboard::SummaryMetadata *metaData = new SummaryMetadata();
+    metaData->set_allocated_plugin_data(pluginData);
+
+    const std::string &filename = fw.LogDir() + "/projector_config.pbtxt";;
+    mlboard::ProjectorConfig *config = new ProjectorConfig();
+
+    // Parse possibly existing config file.
+    std::ifstream fin(filename);
+    if (fin.is_open()) {
+        std::ostringstream ss;
+        ss << fin.rdbuf();
+        google::protobuf::TextFormat::ParseFromString(ss.str(), config);
+        fin.close();
+    }
+
+    mlboard::EmbeddingInfo *embedding = config->add_embeddings();
+    embedding->set_tensor_name(tensorName);
+    embedding->set_tensor_path(tensordataPath);
+    if (metadataPath != "") {
+        embedding->set_metadata_path(metadataPath);
+    }
+    if (tensorShape.size() > 0) {
+        for (size_t shape : tensorShape) embedding->add_tensor_shape(shape);
+    }
+
+    // The `config` and `embedding` pointers will be deleted by the
+    // ProjectorConfig destructor.
+    std::ofstream fout(filename);
+    std::string content;
+    google::protobuf::TextFormat::PrintToString(*config, &content);
+    fout << content;
+    fout.close();
+
+    mlboard::Summary *summary = new Summary();
+    mlboard::Summary_Value *v = summary->add_value();
+    v->set_tag("embedding");
+    v->set_allocated_metadata(metaData);
+
+    fw.CreateEvent(1, summary);
+}
+
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::Embedding(
+      const std::string& tensorName,
+      const arma::mat& tensordata,
+      const std::vector<std::string> &metadata,
+      Filewriter& fw,
+      std::string tensordataPath,
+      std::string metadataPath,
+      std::string relativeTensordataPath,
+      std::string relativeMetadataPath)
+{
+  // Deafult file name.
+  if(tensordataPath == "")
+    tensordataPath = fw.LogDir() + "/tensor.tsv";
+  if(metadataPath == "")
+    metadataPath = fw.LogDir() + "/meta.tsv";
+  std::ofstream tensordDataFile(tensordataPath);
```

Comment:
```suggestion
  std::ofstream tensorDataFile(tensordataPath);
```
I am not sure whether the variable name is selected with some purpose or it's a mistake. Let me know if I am missing something.